### DISCUSSION TITLE
Import attributes support

### DISCUSSION THREAD

**Discussion by lucsky** [rollup/rollup#5299] [Q&A]
Hello, does Rollup support TC39 import attributes (https://github.com/tc39/proposal-import-attributes) which was also recently added to Typescript 5.3 ? The documentation only seems to mention the old deprecated syntax (which uses the `assert` keyword).

**Comment by lukastaegert** [comment]
The syntax is supported, but it is currently rendered as "assert" in the output as the "with" syntax is not widely supported: https://rollupjs.org/repl/?version=4.9.0&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMGZvbyUyMGZyb20lMjAnZXh0ZXJuYWwnJTIwd2l0aCUyMCU3QnR5cGUlM0ElMjAnanNvbiclN0QlM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMm91dHB1dCUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTdEJTJDJTIydHJlZXNoYWtlJTIyJTNBdHJ1ZSU3RCU3RA==

**Comment by Kegulf** [comment]
Exciting! Node 22 drops support for the `assert` keyword.
https://nodejs.org/docs/latest-v22.x/api/esm.html#import-attributes

In other word, Rollup is no longer compatible with Node as of v22

**Reply by Kegulf** [reply]
Ref:
https://github.com/nodejs/node/pull/52104

**Reply by TrickyPi** [reply]
Please see https://rollupjs.org/configuration-options/#output-importattributeskey.

**Reply by Kegulf** [reply]
Hmm, @TrickyPi should the default not be `with` as that is what the TC39 proposal is?

**Reply by Kegulf** [reply]
This is actually a tricky one
Assert support: https://caniuse.com/?search=import%20assertions
`with` support: https://caniuse.com/mdn-javascript_statements_import_import_attributes

**Reply by TrickyPi** [reply]
> Hmm, @TrickyPi should the default not be `with` as that is what the TC39 proposal is?

That is a breaking change for Rollup V4, and we have a plan to do that in V5.

**Reply by Kegulf** [reply]
Hoi @TrickyPi 😁 

Can't seem to get this option to work.. Any idea what I'm doing wrong? Still getting the error.
```SHELL
[!] SyntaxError: Unexpected identifier 'assert'
```

Tried the option both directly in config and in CLI.

```TS
import packageJson from "./package.json" with { type: "json" };

export default {
  input: "./src/index.ts",
  output: [
    {
      file: packageJson.exports["."].import.default,
      format: "es",
      sourcemap: true,
      importAttributesKey: "with",
    },
    {
      file: packageJson.exports["."].require.default,
      format: "cjs",
      sourcemap: true,
      importAttributesKey: "with",
    },
  ],
  .... Some unrelated stuff,
};
```

**Reply by TrickyPi** [reply]
Could you create a minimal reproduction in StackBlitz or GitHub?

**Reply by ryanoglesby08** [reply]
Here is a minimal repro using CodeSandbox (Stackblitz was using Node v18 and didn't seem to let me change it)

https://codesandbox.io/p/devbox/rsrx6s

* Run `npm run build` and see the error.
* The error occurs when using a rollup.config.ts file and `--configPlugin @rollup/plugin-typescript`
* An identical rollup.config.mjs is also included. This one works fine.

**Reply by TrickyPi** [reply]
Thanks for reproducing this. We overlooked the build process for the config file. I'll look into this and fix it.

**Reply by TrickyPi** [reply]
The new option [configImportAttributesKey](https://rollupjs.org/command-line-interface/#configimportattributeskey-with-assert) has been introduced to resolve this issue and has been released in version 4.28.0
